### ISSUE TITLE
Support multiple workflows in the same tree

### ISSUE THREAD

**Issue by algernon** [nvie/gitflow#154]
A request arrived in Debian bug[#642194](http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=642194) to allow one to have multiple workflow setups in the same tree: so that one can use a workflow setup for upstream development, and another for packaging, for example.

My first thought was to add a new function to gitflow-common, that decides what prefix to use for git config calls (defaulting to gitflow, to maintain the status quo), and teach this function to recognise the arbitrarily chosen GIT_FLOW_SELF environment variable (or, lacking that, $0) as a place to extract the prefix from: git-flow is stripped from these, and gitflow prepended.

This way, if one sets GIT_FLOW_SELF to, say, git-flow-debian, then from that point onwards, git flow will look for its config under the gitflow-debian label, instead of gitflow.

Thus, setting the "dflow" alias to "!GIT_FLOW_SELF=git-flow-debian git flow" will work.

I have a patch that implements this, and will send a pull request in a couple of minutes.

**Comment by algernon** [comment]
Pull request [#155](https://github.com/nvie/gitflow/pull/155) implements the changes explained above.

**Comment by jewzaam** [comment]
It would be nice to have a git-flow command to switch between setups.  Thank you for this feature, it's a must for my work stream, as we often have multiple branches of development going on in parallel. Cheers!
